metrics/tenant_registries.go

```diff
@@ -248,7 +265,6 @@ func (d MetricFamiliesPerTenant) foldGauges(out chan<- prometheus.Metric, desc *
 	}
 
 	out <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, result)
-
 }
 
 func (d MetricFamiliesPerTenant) SendMinOfGauges(out chan<- prometheus.Metric, desc *prometheus.Desc, gauge string) {
```

Comment:
This calls d.foldGauges above, which returns 0 as default, so I don't think the statement in the PR description holds: 
> MetricFamiliesPerTenant.SendMinOfGauges will no longer return 0 if any tenant doesn't have the gauge.

Reply:
I think I should fix the message to say "MetricFamiliesPerTenant.SendMinOfGauges will no longer return 0 if **some** tenant doesn't have the gauge."

Reply:
> > MetricFamiliesPerTenant.SendMinOfGauges will no longer return 0 if any tenant doesn't have the gauge.
> 
> This message from changelog describes fixed bug that exists in current version of dskit (before this PR): if there's a tenant registry without a gauge with given name, then `SendMinOfGauges` for given gauge always sends value 0 (assuming there's no value below 0). This PR fixes it.

The test bellow has:
```
	// No matching metric.
	verifySendMinOfGauges(t, 0)
```
It explicitly verifies the opposite, which is the behavior before this PR as far as I can tell. Confused...

Reply:
The test relevant for this changelog entry is on line 326: https://github.com/grafana/dskit/blob/62aaeef8ecad2bb120a5c4c696ccf439c85a7c96/metrics/tenant_registries_test.go#L326

At that point in the test, there are 2 user-registries in `regs`, but only one of them has `test_metric` gauge registered. Now the code correctly sends `100`, before the fix it would send `0`.

In this changelog entry I'm trying to explain this bugfix, but I'm failing :(

Reply:
ok, now I get it. That makes sense. Although that means that to test the same behavior with MAX, you need to set one user's gauge to -100 and not set a gauge on the other , otherwise the bug with the 0 does not show up.